### PR TITLE
Create migration

### LINKED ISSUE #31: [Model] Create migration

### PR DESCRIPTION
**Pull request by mvn-quangtran-dn**
@at-vienlm please help me review this
Fix #31

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -14,9 +14,15 @@ public function up()
     {
         Schema::create('users', function (Blueprint $table) {
             $table->increments('id');
-            $table->string('name');
-            $table->string('email')->unique();
+            $table->string('username')->unique();
+            $table->string('fullname');
             $table->string('password');
+            $table->tinyInteger('gender');
+            $table->date('birthday');
```
Comment (ghost) [inline database/migrations/2014_10_12_000000_create_users_table.php]: can `nullable`

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 2]
```diff
@@ -14,9 +14,15 @@ public function up()
...
+            $table->string('phone');
```
Comment (ghost) [inline database/migrations/2014_10_12_000000_create_users_table.php]: can `nullable`

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 3]
```diff
@@ -14,9 +14,15 @@ public function up()
...
+            $table->string('address');
```
Comment (ghost) [inline database/migrations/2014_10_12_000000_create_users_table.php]: can `nullable`

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 4]
```diff
@@ -14,9 +14,15 @@ public function up()
...
+            $table->date('expiretime');
+            $table->string('image');
```
Comment (ghost) [inline database/migrations/2014_10_12_000000_create_users_table.php]: can `nullable`

File: database/migrations/2016_07_04_065925_create_borrows_table.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    public function up()
+    {
+        Schema::create('borrows', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('user_id')->unsigned();
+            $table->foreign('user_id')->references('id')->on('users')->onDelete('cascade');
+            $table->integer('admin_user_id')->unsigned();
+            $table->foreign('admin_user_id')->references('id')->on('admin_users')->onDelete('cascade');
+            $table->integer('quantity');
```
Comment (ghost) [inline database/migrations/2016_07_04_065925_create_borrows_table.php]: should have default value

File: database/migrations/2016_07_04_070011_create_categories_table.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('categories', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name');
```
Comment (ghost) [inline database/migrations/2016_07_04_070011_create_categories_table.php]: should `unique`

File: database/migrations/2016_07_04_070011_create_categories_table.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            $table->integer('admin_user_id')->unsigned();
+            $table->foreign('admin_user_id')->references('id')->on('admin_users')->onDelete('cascade');
+            $table->integer('quantity');
```
Comment (ghost) [inline database/migrations/2016_07_04_070011_create_categories_table.php]: should have default value

File: database/migrations/2016_07_04_070119_create_books_table.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        Schema::create('books', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name');
+            $table->integer('category_id')->unsigned();
+            $table->foreign('category_id')->references('id')->on('categories')->onDelete('cascade');
+            $table->integer('admin_user_id')->unsigned();
+            $table->foreign('admin_user_id')->references('id')->on('admin_users')->onDelete('cascade');
+            $table->string('author');
+            $table->integer('quantity');
```
Comment (ghost) [inline database/migrations/2016_07_04_070119_create_books_table.php]: should have default value

File: database/migrations/2016_07_04_070119_create_books_table.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            $table->string('image');
```
Comment (ghost) [inline database/migrations/2016_07_04_070119_create_books_table.php]: should nullable

File: database/migrations/2016_07_04_070132_create_book_items_table.php [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('book_items', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('book_id')->unsigned();
+            $table->foreign('book_id')->references('id')->on('books')->onDelete('cascade');
+            $table->integer('admin_user_id')->unsigned();
```
Comment (ghost) [inline database/migrations/2016_07_04_070132_create_book_items_table.php]: I think we don't need `admin_user_id` in this table

File: database/migrations/2016_07_04_065925_create_borrows_table.php (hunk 5 above)
Comment (at-quangth) [inline database/migrations/2016_07_04_065925_create_borrows_table.php]: what field must have default value ? @at-vienlm

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 4 above)
Comment (at-quangth) [inline database/migrations/2014_10_12_000000_create_users_table.php]: an

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 11]
```diff
@@ -14,9 +14,15 @@ public function up()
...
-            $table->string('name');
-            $table->string('email')->unique();
+            $table->string('username')->unique();
+            $table->string('fullname');
             $table->string('password');
+            $table->tinyInteger('gender');
+            $table->date('birthday');
+            $table->string('phone');
+            $table->string('address');
+            $table->date('expiretime');
```
Comment (at-quangth) [inline database/migrations/2014_10_12_000000_create_users_table.php]: anh

**Comment by at-vienlm** [comment]
#31

File: database/migrations/2016_07_04_065925_create_borrows_table.php [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('borrows', function (Blueprint $table) {
+            $table->increments('id',10);
```
Comment (ghost) [inline database/migrations/2016_07_04_065925_create_borrows_table.php]: no need `$table->increments('id',10);`, just `$table->increments('id');`
And please fix all

**Pull request merged**